src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -125,6 +127,19 @@ public function validate($form, Constraint $formConstraint)
                         }
                     }
                 }
+
+                foreach ($form->all() as $field) {
+                    if ($field->isSubmitted()) {
+                        $this->resolvedGroups[$field] = $groups;
+                        $fieldFormConstraint = new Form();
+                        $validator->atPath(sprintf($fieldPropertyPath, $field->getPropertyPath()))->validate($field, $fieldFormConstraint);
+                    }
+                }
+            }
+
+            if ($hasChildren && $form->isRoot()) {
+                // destroy storage to avoid memory leaks
+                $this->resolvedGroups = null;
```

Comment:
maybe we should create a new SplObjectStorage rather than `null` to avoid the issues about illegal offsets ?

Reply:
agreed, it doesn't hurt but can potentially prevent some PHP errors